### PR TITLE
Support for singular resource

### LINKED ISSUE #6: Single resource support
*jsonapi_resource* controller method only generates the documentation for plural resources.
I propose to add single resource support via a new optional argument called singular such as:

```ruby
jsonapi_resource '/v1/user', singular: true
```

I'll work on a pull request to add this feature.

### PR DESCRIPTION
**Pull request by aokiji**
Fix #6

### CODE REVIEW

File: lib/jsonapi_swagger_helpers/readable.rb [hunk 1]
```diff
@@ -1,21 +1,26 @@
...
+                    :controller,
+                    :resource,
+                    :description,
+                    :tags,
+                    :singular
       end
     end
 
-    def initialize(node, controller, description: nil, tags: [])
+    def initialize(node, controller, description: nil, tags: [],
```
Comment (richmolj) [inline lib/jsonapi_swagger_helpers/readable.rb]: can we keep this on the same line?

File: lib/jsonapi_swagger_helpers/resource_mixin.rb [hunk 2]
```diff
@@ -41,7 +44,8 @@ def load_resource(config)
           action_class_name = "#{action.to_s.camelize}Action"
           action_class = JsonapiSwaggerHelpers.const_get(action_class_name)
           action_object = action_class.new \
-            self, controller, tags: tags, description: descriptions[action]
+            self, controller, tags: tags, description: descriptions[action],
+                              singular: singular
```
Comment (richmolj) [inline lib/jsonapi_swagger_helpers/resource_mixin.rb]: Can we keep this on the same line?

File: lib/jsonapi_swagger_helpers/writeable.rb [hunk 3]
```diff
@@ -1,21 +1,26 @@
...
+                    :resource,
+                    :description,
+                    :tags,
+                    :singular
       end
     end
 
-    def initialize(node, controller, description: nil, tags: [])
+    def initialize(node, controller, description: nil, tags: [],
+                   singular: false)
```
Comment (richmolj) [inline lib/jsonapi_swagger_helpers/writeable.rb]: Can we keep this on the same line?

**Review by richmolj — COMMENTED** [review]
Minor formatting comments but looking good, thanks! 👍

File: lib/jsonapi_swagger_helpers/writeable.rb (hunk 3 above)
Comment (aokiji) [reply lib/jsonapi_swagger_helpers/writeable.rb]: Yeah no problem, I was just using rubocop with line limit of 80. It maybe a good idea to use some kind of static analyzer?

Comment (richmolj) [reply lib/jsonapi_swagger_helpers/writeable.rb]: I'd just kick em all down if we hit 80

**Comment by aokiji** [comment]
@richmolj I've fixed the formatting issues

**Comment by richmolj** [comment]
Thanks @aokiji!

**Pull request merged**